extensions/security-jpa/runtime/src/main/java/io/quarkus/security/jpa/runtime/JpaIdentityProvider.java

```diff
@@ -37,16 +39,16 @@ public Uni<SecurityIdentity> authenticate(UsernamePasswordAuthenticationRequest
         return context.runBlocking(new Supplier<SecurityIdentity>() {
             @Override
             public SecurityIdentity get() {
-                EntityManager em = entityManagerFactory.createEntityManager();
-                ((org.hibernate.Session) em).setHibernateFlushMode(FlushMode.MANUAL);
-                ((org.hibernate.Session) em).setDefaultReadOnly(true);
-                try {
-                    return authenticate(em, request);
+                if (requireActiveCDIRequestContext() && !Arc.container().requestContext().isActive()) {
+                    Arc.container().requestContext().activate();
+                }
```

Comment:
I'm not sure you're supposed to do this... In other areas we request that users explicitly add the `@ActivateRequestContext` annotation to their code if they need the request context.

If you still want to do this, you should deactivate the request context after you're done, no? See:

https://github.com/quarkusio/quarkus/blob/ebee0a4b66d1b4dba9c51165e78d22fbf3b90d3e/independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ActivateRequestContextInterceptor.java#L121-L133

Reply:
> I'm not sure you're supposed to do this... In other areas we request that users explicitly add the @ActivateRequestContext annotation to their code if they need the request context.

I'm not happy to active request context, but users can't activate request context here, there is no place how they could do it on generated beans (and it wouldn't be on non-generated anyway unless they were supposed to provide their implementation).

> If you still want to do this, you should deactivate the request context after you're done, no? See:

Good point, thank you.